### PR TITLE
Implement `np.roots`

### PR DESCRIPTION
**Pull request by stuartarchibald**
Implements and tests the `np.roots` function.

Intended to close:
https://github.com/numba/numba/issues/2062

### CODE REVIEW

File: numba/targets/polynomial.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        cast_t = np.float64
+    else:
+        cast_t = np_support.as_dtype(ty)
+
+    def roots_impl(p):
+        # impl based on numpy:
+        # https://github.com/numpy/numpy/blob/master/numpy/lib/polynomial.py
+
+        if len(p.shape) != 1:
+            raise ValueError("Input must be a rank-1 array.")
```
Comment (pitrou) [inline numba/targets/polynomial.py]: "1-d" or "1d" would be the conventional wording.

Comment (stuartarchibald) [inline numba/targets/polynomial.py]: That's mirroring numpy, want me to change it?

Comment (pitrou) [inline numba/targets/polynomial.py]: I think that would be better, yes.

Comment (stuartarchibald) [inline numba/targets/polynomial.py]: Sure, fixed in 	f91b1de.

**Pull request merged**